app/page.tsx

```
@@ -95,6 +99,43 @@ function ChatPageContent() {
 
   const handleTopUp = (_amount?: number) => {};
 
+  // Capture prefilled topup token from URL and remove it from the address bar
+  useEffect(() => {
+    if (!topupFromUrl) return;
+    try {
+      localStorage.setItem('prefilled_topup_token', topupFromUrl);
+    } catch {}
```

Can we log the error instead of silently catching?